File: android/feature/ministros/src/main/java/com/escala/ministerial/feature/ministros/domain/model/Ministro.kt
package com.escala.ministerial.feature.ministros.domain.model

import java.time.LocalDate

data class Ministro(
    val id: Long,
    val nome: String,
    val email: String,
    val telefone: String?,
    val dataNascimento: LocalDate?,
    val observacoes: String?,
    val ativo: Boolean,
    val visitasAoInfermo: Boolean,
    val statusCurso: Boolean,
    val escalasMes: Int,
    val funcao: FuncaoMinistro,
    val funcaoEspecificada: String? = null,
    val escalasAgendadas: List<LocalDate> = emptyList(),
    val indisponibilidades: List<Indisponibilidade> = emptyList(),
)

enum class FuncaoMinistro(val label: String) {
    EUCARISTIA("Eucaristia"),
    LEITURA("Leitura"),
    ACOLHIMENTO("Acolhimento"),
    MUSICA("Música"),
    CATEQUESE("Catequese"),
    ADORACAO("Adoração"),
    OUTRO("Outro"),
}

File: ios/EscalaMinisterial/Models/Ministro.swift
import Foundation

struct Ministro: Codable, Identifiable {
    var id: Int?
    var nome: String
    var email: String
    var telefone: String?
    var dataNascimento: String?
    var observacoes: String?
    var ativo: Bool
    var funcao: String
    var funcaoEspecificada: String?
    var visitasAoInfermo: Bool
    var statusCurso: Bool
    var escalasMes: Int?
    var indisponibilidades: [Indisponibilidade]?
    var escalasAgendadas: [String]?

    enum FuncaoMinistro: String, CaseIterable, Codable {
        case eucaristia = "EUCARISTIA"
        case leitura = "LEITURA"
        case acolhimento = "ACOLHIMENTO"
        case musica = "MUSICA"
        case catequese = "CATEQUESE"
        case adoracao = "ADORACAO"
        case outro = "OUTRO"

        var label: String {
            switch self {
            case .eucaristia:  return "Eucaristia"
            case .leitura:     return "Leitura"
            case .acolhimento: return "Acolhimento"
            case .musica:      return "Música"
            case .catequese:   return "Catequese"
            case .adoracao:    return "Adoração"
            case .outro:       return "Outro"
            }
        }
    }

    var funcaoLabel: String {
        if funcao == FuncaoMinistro.outro.rawValue, let esp = funcaoEspecificada, !esp.isEmpty {
            return esp
        }
        return FuncaoMinistro(rawValue: funcao)?.label ?? funcao
    }
}
